opts/envdir_test.go

```diff
@@ -0,0 +1,65 @@
+package opts
+
+import (
+	"testing"
+)
+
+func TestParseEnvDir(t *testing.T) {
+	fixtures := []struct {
+		dir     string
+		results []string
+		failure bool
+	}{
+		{"fixtures/envdir/simple", []string{"BAR=baz", "FOO=bar"}, false},
+		{"fixtures/envdir/firstline", []string{"FOO=bar"}, false},
+		{"fixtures/envdir/junk", []string{"FOO=bar"}, false},
+		{"fixtures/envdir/trim", []string{"FOO=bar"}, false},
+		{"fixtures/envdir/null", []string{"FOO=ba\nr"}, false},
+		{"fixtures/envdir/empty", []string{}, false},
+		{"fixtures/envdir/whitespace", []string{}, true},
```

Comment:
did you forget to add the "whitespace" dir?
If you meant for it to test a missing dir, use a better name like "nodir".
Then add a 'whitespace' dir to test bad file names.